### ISSUE TITLE
Recent change to package-lock.json path breaks npm workspaces

### ISSUE THREAD

**Issue by chrisui** [floydspace/serverless-esbuild#202]
[This change from 5 days ago](https://github.com/floydspace/serverless-esbuild/commit/9d81dce7105af5ebb69d0dc42a383d0bcd8e1aa3) has broken our builds because with **npm workspaces** we only have a `project-lock.json` at the root.

Perhaps this path needs to be variable based on the packager configured.

_Originally posted by @chrisui in https://github.com/floydspace/serverless-esbuild/issues/199#issuecomment-930106781_

**Comment by olup** [comment]
Yeah, so I made those changes, and I think we can find a middle ground.

Detection of a local package.json was the logic implemented to detect the use of monorepo. But because it was sought at root level, detection would always fail, what I did was to go back to a local detection. See, it is because I am using a monorepo with workspaces and because I need inclusion of externals that in my case installation with frozen lockfile would throw.

I think I know what your problem would be -  but for the sake of drafting a PR that makes sense in every situation, could you please elaborate on your setup? What exactly in your build is broken ? are you using individual packaging? Do you have external libraries listed?
